public/components/trace_analytics/components/common/shared_components/custom_datagrid.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useMemo } from 'react';
+import {
+  EuiDataGrid,
+  EuiButtonEmpty,
+  EuiDataGridColumn,
+  EuiDataGridSorting,
+  EuiOverlayMask,
+  EuiButtonIcon,
+} from '@elastic/eui';
+import { NoMatchMessage } from '../helper_functions';
+
+interface FullScreenWrapperProps {
+  children: React.ReactNode;
+  onClose: () => void;
+  isFullScreen: boolean;
+}
+
+// EUI Data grid full screen button is currently broken, this is a workaround
+const FullScreenWrapper: React.FC<FullScreenWrapperProps> = ({
+  children,
+  onClose,
+  isFullScreen,
+}) => {
+  if (!isFullScreen) return <>{children}</>;
+
+  return (
+    <EuiOverlayMask>
+      <div className="full-screen-wrapper">
+        <EuiButtonIcon
+          iconType="cross"
+          aria-label="Close full screen"
+          onClick={onClose}
+          display="empty"
+          className="full-screen-close-icon"
+        />
+        <div className="full-screen-content">{children}</div>
+      </div>
+    </EuiOverlayMask>
+  );
+};
+
+interface PaginationParams {
+  pageIndex: number;
+  pageSize: number;
+  pageSizeOptions?: number[];
+  onChangePage: (page: number) => void;
+  onChangeItemsPerPage: (size: number) => void;
+}
+
+interface RenderCustomDataGridParams {
+  columns: EuiDataGridColumn[];
+  renderCellValue: (props: any) => React.ReactNode;
```

Comment:
We can remove the use of any here and re-use the actual props extended with `disableInteractions`

Reply:
Switched too 
```
renderCellValue: (props: RenderCellValueProps) => React.ReactNode;

interface RenderCellValueProps {
  rowIndex: number;
  columnId: string;
  disableInteractions: boolean;
  ```